src/test/java/com/uwetrottmann/tmdb2/TmdbTestSuite.java

```diff
@@ -0,0 +1,291 @@
+package com.uwetrottmann.tmdb2;
+
+import com.googlecode.junittoolbox.SuiteClasses;
+import com.googlecode.junittoolbox.WildcardPatternSuite;
+import com.uwetrottmann.tmdb2.annotations.RequiresAccountSession;
+import com.uwetrottmann.tmdb2.annotations.RequiresGuestSession;
+import com.uwetrottmann.tmdb2.entities.Account;
+import com.uwetrottmann.tmdb2.entities.BaseMovie;
+import com.uwetrottmann.tmdb2.entities.FavoriteMedia;
+import com.uwetrottmann.tmdb2.entities.ListCreateRequest;
+import com.uwetrottmann.tmdb2.entities.ListCreateResponse;
+import com.uwetrottmann.tmdb2.entities.ListOperation;
+import com.uwetrottmann.tmdb2.entities.RatingObject;
+import com.uwetrottmann.tmdb2.entities.WatchlistMedia;
+import com.uwetrottmann.tmdb2.enumerations.AuthenticationType;
+import com.uwetrottmann.tmdb2.enumerations.MediaType;
+import org.assertj.core.util.Strings;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.ExternalResource;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Vector;
+
+import static com.uwetrottmann.tmdb2.BaseTestCase.testClassToRun;
+import static com.uwetrottmann.tmdb2.TestData.initializeTestingIntegrityData;
+import static com.uwetrottmann.tmdb2.TestData.testList;
+import static com.uwetrottmann.tmdb2.TestData.testListMovies;
+import static com.uwetrottmann.tmdb2.TestData.testMovie;
+import static com.uwetrottmann.tmdb2.TestData.testTvEpisode;
+import static com.uwetrottmann.tmdb2.TestData.testTvSeason;
+import static com.uwetrottmann.tmdb2.TestData.testTvShow;
+
+@RunWith(WildcardPatternSuite.class)
+@SuiteClasses({"**/*.class", "!TmdbTestSuite.class", "!BaseTestCase.class"})
+public class TmdbTestSuite {
+
+    private static String API_KEY;
+    private static String USERNAME;
+    private static String PASSWORD;
+    static final boolean DEBUG = true;
+
+    static Tmdb unauthenticatedInstance;
+    static Tmdb authenticatedInstance;
+
+
+    public static Boolean initialized = false;
+    public static Boolean accountDataInitialized = false;
+    public static Boolean guestDataInitialized = false;
+
+    public static Boolean suiteRunning = false;
+
+    private static Account account;
+
+    @ClassRule
+    public static ExternalResource triggerSuiteRunning = new ExternalResource() {
+        @Override
+        protected void before() throws Throwable {
+            suiteRunning = true;
+        }
+    };
+
+    @BeforeClass
+    public static void initializeEnvironment() throws IOException, ParseException, InterruptedException {
+        if (initialized) {
+            return;
+        }
+
+        if (Strings.isNullOrEmpty(USERNAME) || Strings.isNullOrEmpty(PASSWORD)) {
+            USERNAME = System.getenv("TMDB_USERNAME");
```

Comment:
Can't this be solved using a properties file? Setting an environment variable seems overkill. Or would a travis build not work then?